libpod/runtime_pod_linux.go

```diff
@@ -133,7 +133,7 @@ func (r *Runtime) NewPod(ctx context.Context, p specgen.PodSpecGenerator, option
 			pod.config.Name = name
 		}
 
-		if p.InfraContainerSpec != nil && p.InfraContainerSpec.Hostname == "" {
+		if p.InfraContainerSpec != nil && p.InfraContainerSpec.Hostname == "" && (p.InfraContainerSpec.UtsNS.NSMode == "private" || p.InfraContainerSpec.UtsNS.NSMode == "") {
```

Comment:
Please use the constants to match this, you could just use UtsNs.IsPrivate().
Also checking this here will not work, compare https://github.com/containers/podman/pull/14128